src/components/definition-list/dl.component.js

```diff
@@ -3,68 +3,68 @@ import propTypes from "@styled-system/prop-types";
 import PropTypes from "prop-types";
 import { StyledDl, StyledDtDiv, StyledDdDiv } from "./definition-list.style";
 import Dt from "./dt.component";
-import Dd from "./dd.component";
 
 const Dl = ({
   children,
   w = 50,
   dtTextAlign = "right",
   ddTextAlign = "left",
-  ...props
+  ...rest
 }) => {
-  const dlComponent = [];
   const listChildren = React.Children.toArray(children);
-  let key = "";
 
-  const composeDlComponent = (array) => {
-    let dtLabel;
-    let ddContent = [];
-    let isLastChild;
-    let nextItemIsNotDd;
+  const wrapChild = (child, key) => {
+    let wrappedChild;
 
-    array.forEach((child, index) => {
-      if (child.type === React.Fragment) {
-        composeDlComponent(child.props.children);
-      } else {
-        if (child.type === Dt) {
-          dtLabel = child;
-        }
-        if (child.type === Dd) {
-          ddContent.push(child);
-        }
+    if (child?.type === Dt) {
+      wrappedChild = (
+        <StyledDtDiv key={key} dtTextAlign={dtTextAlign}>
+          {child}
+        </StyledDtDiv>
+      );
+    } else {
+      wrappedChild = (
+        <StyledDdDiv key={key} ddTextAlign={ddTextAlign}>
+          {child}
+        </StyledDdDiv>
+      );
+    }
 
-        isLastChild = index === listChildren.length - 1;
-        nextItemIsNotDd =
-          !isLastChild &&
-          [Dt, React.Fragment].includes(listChildren[index + 1].type);
+    return wrappedChild;
+  };
 
-        if (dtLabel && (nextItemIsNotDd || isLastChild)) {
-          key = `${key + 1}`;
+  const composeContent = (childArray) => {
+    const contentArray = [];
 
-          dlComponent.push(
-            <React.Fragment key={child.props.key || key}>
-              <StyledDtDiv dtTextAlign={dtTextAlign}>{dtLabel}</StyledDtDiv>
-              <StyledDdDiv ddTextAlign={ddTextAlign}>{ddContent}</StyledDdDiv>
-            </React.Fragment>
+    childArray.forEach((child) => {
+      if (child?.type === React.Fragment) {
+        const fragmentList = [];
+        child.props.children.forEach((fragmentChild, fragmentIndex) => {
```

Comment:
Could this not be a `map`?

```suggestion
        const fragmentList = child.props.children.map((fragmentChild, fragmentIndex) => (
          wrapChild(fragmentChild, contentArray.length + fragmentIndex)
        ));
```

Reply:
Yeah it can, looks like I forgot to simplify it when I had it working